src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -615,37 +615,73 @@ public static string ExtractLettersAndNumbers(string value)
             return Regex.Replace(value, "[^A-Za-z0-9]", "");
         }
 
+        /// <summary>
+        /// Checks that an input IList is monotonically increasing.
+        /// </summary>
+        /// <param name="values">An array of values</param>
+        /// <returns>True if the array is monotonically increasing; false otherwise.
+        /// ILists containing NaN values are considered to be not monotonically increasing.</returns>
         public static bool IsSorted(IList<float> values)
         {
             if (Utils.Size(values) <= 1)
                 return true;
 
-            var prev = values[0];
-
-            for (int i = 1; i < values.Count; i++)
+            var previousValue = values[0];
+            var listLength = values.Count;
+            for (int i = 1; i < listLength; i++)
             {
-                if (!(values[i] >= prev))
+                var currentValue = values[i];
+                if (currentValue < previousValue)
```

Comment:
I actually disagree about `Sort()`. I think that the presence of NaNs should result in a return of `false`. The assumption of ordered arrays in our codebase is that the numbers in the array are monotonically increasing. NaN is not a number, so it cannot be part of a monotonically increasing sequence.

And practically speaking, these methods are used as sanity checks on user-provided data, so we don't want to make them more permissive than they currently are.

Reply:
Also, the original functions were set to fail out on NaNs. I do not want to change the behavior of these functions because I don't know what the downstream effects will be.

Reply:
Then let's rename the method to IsMonotonicallyIncreasing

---
In reply to: [250355488](https://github.com/dotnet/machinelearning/pull/2188#discussion_r250355488) [](ancestors = 250355488)

Reply:
Otherwise need to rename the method.

---
In reply to: [250036842](https://github.com/dotnet/machinelearning/pull/2188#discussion_r250036842) [](ancestors = 250036842)

Reply:
Renamed.